lib/src/metta/runner/stdlib.rs

```diff
@@ -1012,6 +948,302 @@ impl Grounded for MatchOp {
+            self.metta.load_module_at_path(path, None).map_err(|e| ExecError::from(e))?;
+
+            unit_result()
+        }
+
+        fn match_(&self, other: &Atom) -> MatchResultIter {
+            match_by_equality(self, other)
+        }
+    }
+
+    /// Provides access to module in a remote git repo, from within MeTTa code
+    /// Similar to `register-module!`, this op will bypass the catalog search
+    #[derive(Clone, Debug)]
+    pub struct GitModuleOp {
+        //TODO-HACK: This is a terrible horrible ugly hack that should be fixed ASAP
+        context: std::sync::Arc<std::sync::Mutex<Vec<std::sync::Arc<std::sync::Mutex<&'static mut RunContext<'static, 'static, 'static>>>>>>,
+    }
+
+    impl PartialEq for GitModuleOp {
+        fn eq(&self, _other: &Self) -> bool { true }
+    }
+
+    impl GitModuleOp {
+        pub fn new(metta: Metta) -> Self {
+            Self{ context: metta.0.context.clone() }
+        }
+    }
+
+    impl Display for GitModuleOp {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            write!(f, "git-module!")
+        }
+    }
+
+    impl Grounded for GitModuleOp {
+        fn type_(&self) -> Atom {
+            Atom::expr([ARROW_SYMBOL, ATOM_TYPE_ATOM, UNIT_TYPE()])
+        }
+
+        fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+            let arg_error = "git-module! expects a URL; use quotes if needed";
+            let url_arg_atom = args.get(0).ok_or_else(|| ExecError::from(arg_error))?;
+            // TODO: When we figure out how to address varargs, it will be nice to take an optional branch name
+
+            let url = match url_arg_atom {
+                Atom::Symbol(url_arg) => url_arg.name(),
+                Atom::Grounded(g) => g.downcast_ref::<Str>().ok_or_else(|| ExecError::from(arg_error))?.as_str(),
+                _ => return Err(arg_error.into()),
+            };
+            let url = strip_quotes(url);
+
+            // TODO: Depending on what we do with `register-module!`, we might want to let the
+            // caller provide an optional mod_name here too, rather than extracting it from the url
+            let mod_name = match mod_name_from_url(url) {
+                Some(mod_name) => mod_name,
+                None => return Err(ExecError::from("git-module! error extracting module name from URL"))
+            };
+
+            let ctx_ref = self.context.lock().unwrap().last().unwrap().clone();
+            let mut context = ctx_ref.lock().unwrap();
+
+            let git_mod_location = ModuleGitLocation::new(url.to_string());
+
+            if let Some((loader, descriptor)) = git_mod_location.get_loader_in_explicit_catalog(&mod_name, UpdateMode::TryFetchLatest, context.metta.environment()).map_err(|e| ExecError::from(e))? {
+                context.get_or_init_module_with_descriptor(&mod_name, descriptor, loader).map_err(|e| ExecError::from(e))?;
+            }
```

Comment:
See my comment above https://github.com/trueagi-io/hyperon-experimental/pull/691#discussion_r1627293050 does it solve this?

Reply:
A special API is still needed in some form because git-loading is a native feature of the PkgInfo.  I could change code so the method was on the Catalog object, which would decouple the Git code from the Environment code.  That would be a little cleaner.

Reply:
I made the API slightly more orthogonal in 46f1df0.  Now GitCatalog doesn't need to know about the environment's explicit_git_mods field.  Although the two places that deal with explicitly loading git modules still need to.